Library/Homebrew/utils/shell.rb

```diff
@@ -163,7 +163,7 @@ def shell_with_prompt(type, preferred_path:, notice:, home: Dir.home)
         zdotdir.mkpath
         FileUtils.chmod_R(0700, zdotdir)
         FileUtils.cp(HOMEBREW_LIBRARY_PATH/"utils/zsh/brew-sh-prompt-zshrc.zsh", zdotdir/".zshrc")
-        %w[.zcompdump .zsh_history .zsh_sessions].each do |file|
+        %w[.zshenv .zcompdump .zsh_history .zsh_sessions].each do |file|
```

Comment:
Also, I added `.zshenv` here to ensure that folks who set things in `.zshenv` get them here, too